### PR TITLE
[Part 2][IT Tests] JAMES-2717 Replace EmbeddedElasticSearchExtension

### PR DESCRIPTION
**Pull request by trantienduchn**
Run all tests and they all pass, except the ESReporterTest with this endless log. Need to investigate
```
[ERROR] c.l.e.m.ElasticsearchReporter - Reporting returned code -1 null: {}
```

### CODE REVIEW

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/ESReporterTest.java [hunk 1]
```diff
@@ -60,18 +61,17 @@
...
     static JamesServerExtension testExtension = new JamesServerBuilder()
-        .extension(embeddedElasticSearchExtension)
+        .extension(elasticSearchExtension)
         .extension(new CassandraExtension())
         .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
             .combineWith(ALL_BUT_JMX_CASSANDRA_MODULE)
             .overrideWith(binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class))
             .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
-            .overrideWith(new TestEmbeddedESMetricReporterModule()))
+            .overrideWith(new TestDockerESMetricReporterModule(elasticSearchExtension.getDockerES().getTcpHost())))
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/ESReporterTest.java]: You need to be using the Http Port, not the Tcp one, no?

**Comment by trantienduchn** [comment]
```
[5e50eac8ccc1bd8d6114685ef4eeb867766c6342] [ERROR] Failures: 
[5e50eac8ccc1bd8d6114685ef4eeb867766c6342] [ERROR]   FixingGhostMailboxTest.ghostMailboxBugShouldDiscardOldContent:205 1 expectation failed.
```
I'm doubling about docker elastichsearch flush (https://github.com/linagora/james-project/pull/2347/commits/38aa5459622279f8d8d21edcde72eb8de9f886be). Anyway, test this please

**Comment by trantienduchn** [comment]
SpamAssassinContract. test this please

**Comment by trantienduchn** [comment]
```
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Apache James :: Server :: JMAP :: Cassandra Integration testing SUCCESS [05:46 min]
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Apache James :: Server :: JMAP :: Memory Integration testing SUCCESS [04:17 min]
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Apache James :: Server :: JMAP :: RabbitMQ + Object Store + Cassandra Integration testing SUCCESS [08:45 min]
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Apache James :: Server :: Web Admin server integration tests SUCCESS [03:23 min]
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Apache James :: Server :: Web Admin 3.4.0-SNAPSHOT . SUCCESS [  0.606 s]
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] ------------------------------------------------------------------------
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] BUILD SUCCESS
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] ------------------------------------------------------------------------
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Total time: 02:22 h
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Finished at: 2019-05-08T18:19:54Z
[f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] ------------------------------------------------------------------------
```

**Comment by chibenwa** [comment]
```
[INFO] apache-james-backends-es ........................... SUCCESS [ 38.961 s]
[INFO] Apache James :: Mailbox :: ElasticSearch ........... SUCCESS [04:47 min]
[INFO] Apache James :: Mailbox :: Plugin :: Quota Search :: ElasticSearch SUCCESS [ 18.939 s][f98edf2768598babbe66dd9fe57f6df70630e14f] [INFO] Apache James :: Server :: JMAP :: Cassandra Integration testing SUCCESS [05:46 min]
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**